storage/local/persistence.go

```diff
@@ -247,11 +252,29 @@ func newPersistence(
 				Help:      "Quantiles for batch indexing duration in seconds.",
 			},
 		),
-		checkpointDuration: prometheus.NewGauge(prometheus.GaugeOpts{
+		checkpointLastDuration: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "checkpoint_last_duration_seconds",
+			Help:      "The duration in seconds it took to last checkpoint in-memory metrics and head chunks.",
+		}),
+		checkpointDuration: prometheus.NewSummary(prometheus.SummaryOpts{
 			Namespace: namespace,
 			Subsystem: subsystem,
 			Name:      "checkpoint_duration_seconds",
-			Help:      "The duration in seconds it took to checkpoint in-memory metrics and head chunks.",
+			Help:      "The duration in seconds taken for checkpointing",
```

Comment:
But in the meantime, you will have very confusing quantile metrics. Setting the quantiles explicitly to empty as suggested works with both versions of client_golang and is least confusing.